From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Salvo Giangreco <giangrecosalvo9@gmail.com>
Date: Sat, 30 May 2026 22:39:08 +0200
Subject: [PATCH] Nuke Knox DualDAR

Refresh for the current SecSettings dex layout. Keep this patch focused
on active DualDAR feature and policy queries so it applies cleanly on the
current S918BXXSAFZE1 base.

---
 14 files changed, 22 insertions(+), 210 deletions(-)

diff --git a/smali_classes2/com/android/settings/password/ChooseLockGeneric$ChooseLockGenericFragment.smali b/smali_classes2/com/android/settings/password/ChooseLockGeneric$ChooseLockGenericFragment.smali
--- a/smali_classes2/com/android/settings/password/ChooseLockGeneric$ChooseLockGenericFragment.smali
+++ b/smali_classes2/com/android/settings/password/ChooseLockGeneric$ChooseLockGenericFragment.smali
@@ -2722,3 +2722 @@
-    invoke-static {}, Lcom/samsung/android/knox/dar/ddar/DualDarManager;->isOnDeviceOwnerEnabled()Z
-
-    move-result v16
+    const/16 v16, 0x0
@@ -2789,3 +2787 @@
-    invoke-static {}, Lcom/samsung/android/knox/dar/ddar/DualDarManager;->isOnDeviceOwnerEnabled()Z
-
-    move-result v0
+    const/4 v0, 0x0
@@ -3904,3 +3900 @@
-    invoke-static {}, Lcom/samsung/android/knox/dar/ddar/DualDarManager;->isOnDeviceOwnerEnabled()Z
-
-    move-result v3
+    const/4 v3, 0x0
@@ -8798,3 +8792 @@
-    invoke-static {v10}, Lcom/samsung/android/knox/SemPersonaManager;->isDarDualEncryptionEnabled(I)Z
-
-    move-result v3
+    const/4 v3, 0x0
diff --git a/smali_classes2/com/android/settings/password/ChooseLockPassword$ChooseLockPasswordFragment.smali b/smali_classes2/com/android/settings/password/ChooseLockPassword$ChooseLockPasswordFragment.smali
--- a/smali_classes2/com/android/settings/password/ChooseLockPassword$ChooseLockPasswordFragment.smali
+++ b/smali_classes2/com/android/settings/password/ChooseLockPassword$ChooseLockPasswordFragment.smali
@@ -2678,3 +2678 @@
-    invoke-static {}, Lcom/samsung/android/knox/dar/ddar/DualDarManager;->isOnDeviceOwnerEnabled()Z
-
-    move-result v3
+    const/4 v3, 0x0
diff --git a/smali_classes2/com/android/settings/password/ConfirmDeviceCredentialBaseFragment.smali b/smali_classes2/com/android/settings/password/ConfirmDeviceCredentialBaseFragment.smali
--- a/smali_classes2/com/android/settings/password/ConfirmDeviceCredentialBaseFragment.smali
+++ b/smali_classes2/com/android/settings/password/ConfirmDeviceCredentialBaseFragment.smali
@@ -668,3 +668 @@
-    invoke-static {}, Lcom/samsung/android/knox/dar/ddar/DualDarManager;->isOnDeviceOwnerEnabled()Z
-
-    move-result v0
+    const/4 v0, 0x0
diff --git a/smali_classes2/com/android/settings/password/ConfirmLockPassword$ConfirmLockPasswordFragment.smali b/smali_classes2/com/android/settings/password/ConfirmLockPassword$ConfirmLockPasswordFragment.smali
--- a/smali_classes2/com/android/settings/password/ConfirmLockPassword$ConfirmLockPasswordFragment.smali
+++ b/smali_classes2/com/android/settings/password/ConfirmLockPassword$ConfirmLockPasswordFragment.smali
@@ -575,3 +575 @@
-    invoke-static {}, Lcom/samsung/android/knox/dar/ddar/DualDarManager;->isOnDeviceOwnerEnabled()Z
-
-    move-result v0
+    const/4 v0, 0x0
@@ -2710,3 +2708 @@
-    invoke-static {}, Lcom/samsung/android/knox/dar/ddar/DualDarManager;->isOnDeviceOwnerEnabled()Z
-
-    move-result v1
+    const/4 v1, 0x0
diff --git a/smali_classes2/com/android/settings/password/ConfirmLockPassword.smali b/smali_classes2/com/android/settings/password/ConfirmLockPassword.smali
--- a/smali_classes2/com/android/settings/password/ConfirmLockPassword.smali
+++ b/smali_classes2/com/android/settings/password/ConfirmLockPassword.smali
@@ -270,3 +270 @@
-    invoke-static {}, Lcom/samsung/android/knox/dar/ddar/DualDarManager;->isOnDeviceOwnerEnabled()Z
-
-    move-result v1
+    const/4 v1, 0x0
@@ -382,3 +380 @@
-    invoke-static {}, Lcom/samsung/android/knox/dar/ddar/DualDarManager;->isOnDeviceOwnerEnabled()Z
-
-    move-result v5
+    const/4 v5, 0x0
diff --git a/smali_classes2/com/android/settings/security/LockUnificationPreferenceController.smali b/smali_classes2/com/android/settings/security/LockUnificationPreferenceController.smali
--- a/smali_classes2/com/android/settings/security/LockUnificationPreferenceController.smali
+++ b/smali_classes2/com/android/settings/security/LockUnificationPreferenceController.smali
@@ -164,3 +164 @@
-    invoke-static {v0}, Lcom/samsung/android/knox/SemPersonaManager;->isDarDualEncryptionEnabled(I)Z
-
-    move-result p0
+    const/4 p0, 0x0
diff --git a/smali_classes4/com/samsung/android/knox/container/KnoxContainerManager.smali b/smali_classes4/com/samsung/android/knox/container/KnoxContainerManager.smali
--- a/smali_classes4/com/samsung/android/knox/container/KnoxContainerManager.smali
+++ b/smali_classes4/com/samsung/android/knox/container/KnoxContainerManager.smali
@@ -3470,38 +3469,0 @@
-    if-nez v0, :cond_1
-
-    monitor-enter p0
-
-    :try_start_0
-    iget-object v0, p0, Lcom/samsung/android/knox/container/KnoxContainerManager;->mDualDARPolicy:Lcom/samsung/android/knox/ddar/DualDARPolicy;
-
-    if-nez v0, :cond_0
-
-    new-instance v0, Lcom/samsung/android/knox/ddar/DualDARPolicy;
-
-    iget-object v1, p0, Lcom/samsung/android/knox/container/KnoxContainerManager;->mContextInfo:Lcom/samsung/android/knox/ContextInfo;
-
-    invoke-direct {v0, v1}, Lcom/samsung/android/knox/ddar/DualDARPolicy;-><init>(Lcom/samsung/android/knox/ContextInfo;)V
-
-    iput-object v0, p0, Lcom/samsung/android/knox/container/KnoxContainerManager;->mDualDARPolicy:Lcom/samsung/android/knox/ddar/DualDARPolicy;
-
-    goto :goto_0
-
-    :catchall_0
-    move-exception v0
-
-    goto :goto_1
-
-    :cond_0
-    :goto_0
-    monitor-exit p0
-
-    return-object v0
-
-    :goto_1
-    monitor-exit p0
-    :try_end_0
-    .catchall {:try_start_0 .. :try_end_0} :catchall_0
-
-    throw v0
-
-    :cond_1
diff --git a/smali_classes4/com/samsung/android/knox/ddar/DualDARPolicy.smali b/smali_classes4/com/samsung/android/knox/ddar/DualDARPolicy.smali
--- a/smali_classes4/com/samsung/android/knox/ddar/DualDARPolicy.smali
+++ b/smali_classes4/com/samsung/android/knox/ddar/DualDARPolicy.smali
@@ -101 +101 @@
-    const-string v0, "1.8.0"
+    const/4 v0, 0x0
@@ -134 +134 @@
-    const/4 v0, 0x1
+    const/4 v0, 0x0
diff --git a/smali_classes5/com/samsung/android/settings/deviceinfo/softwareinfo/KnoxVersionPreferenceController.smali b/smali_classes5/com/samsung/android/settings/deviceinfo/softwareinfo/KnoxVersionPreferenceController.smali
--- a/smali_classes5/com/samsung/android/settings/deviceinfo/softwareinfo/KnoxVersionPreferenceController.smali
+++ b/smali_classes5/com/samsung/android/settings/deviceinfo/softwareinfo/KnoxVersionPreferenceController.smali
@@ -361,3 +361 @@
-    invoke-static {}, Lcom/samsung/android/knox/dar/ddar/DualDarManager;->isOnDeviceOwnerEnabled()Z
-
-    move-result v7
+    const/4 v7, 0x0
diff --git a/smali_classes5/com/samsung/android/settings/knox/KnoxUtils.smali b/smali_classes5/com/samsung/android/settings/knox/KnoxUtils.smali
--- a/smali_classes5/com/samsung/android/settings/knox/KnoxUtils.smali
+++ b/smali_classes5/com/samsung/android/settings/knox/KnoxUtils.smali
@@ -1907,6 +1906,0 @@
-    invoke-static {}, Lcom/samsung/android/knox/dar/ddar/DualDarManager;->isOnDeviceOwnerEnabled()Z
-
-    move-result v0
-
-    if-nez v0, :cond_0
-
@@ -1914,15 +1907,0 @@
-
-    return p0
-
-    :cond_0
-    new-instance v0, Lcom/android/internal/widget/LockPatternUtils;
-
-    invoke-direct {v0, p0}, Lcom/android/internal/widget/LockPatternUtils;-><init>(Landroid/content/Context;)V
-
-    invoke-virtual {v0}, Lcom/android/internal/widget/LockPatternUtils;->getLockPatternUtilForDualDarDo()Lcom/android/internal/widget/LockPatternUtils$LockPatternUtilForDualDarDo;
-
-    move-result-object p0
-
-    invoke-virtual {p0}, Lcom/android/internal/widget/LockPatternUtils$LockPatternUtilForDualDarDo;->getInnerAuthUserForDo()I
-
-    move-result p0
@@ -3090,19 +3068,0 @@
-    invoke-static {p0}, Lcom/samsung/android/knox/EnterpriseDeviceManager;->getInstance(Landroid/content/Context;)Lcom/samsung/android/knox/EnterpriseDeviceManager;
-
-    move-result-object p0
-
-    if-eqz p0, :cond_0
-
-    invoke-virtual {p0}, Lcom/samsung/android/knox/EnterpriseDeviceManager;->getPasswordPolicy()Lcom/samsung/android/knox/devicesecurity/PasswordPolicy;
-
-    move-result-object p0
-
-    if-eqz p0, :cond_0
-
-    invoke-virtual {p0}, Lcom/samsung/android/knox/devicesecurity/PasswordPolicy;->isChangeRequestedForInner()I
-
-    move-result p0
-
-    return p0
-
-    :cond_0
@@ -3117,6 +3076,0 @@
-    invoke-static {}, Lcom/samsung/android/knox/dar/ddar/DualDarManager;->isOnDeviceOwnerEnabled()Z
-
-    move-result v0
-
-    if-nez v0, :cond_0
-
@@ -3124,15 +3077,0 @@
-
-    return p0
-
-    :cond_0
-    new-instance v0, Lcom/android/internal/widget/LockPatternUtils;
-
-    invoke-direct {v0, p0}, Lcom/android/internal/widget/LockPatternUtils;-><init>(Landroid/content/Context;)V
-
-    invoke-virtual {v0}, Lcom/android/internal/widget/LockPatternUtils;->getLockPatternUtilForDualDarDo()Lcom/android/internal/widget/LockPatternUtils$LockPatternUtilForDualDarDo;
-
-    move-result-object p0
-
-    invoke-virtual {p0, p1}, Lcom/android/internal/widget/LockPatternUtils$LockPatternUtilForDualDarDo;->isInnerAuthUserForDo(I)Z
-
-    move-result p0
@@ -3146,58 +3085,3 @@
-    invoke-static {p0}, Landroid/os/UserManager;->get(Landroid/content/Context;)Landroid/os/UserManager;
-
-    move-result-object p0
-
-    if-eqz p0, :cond_2
-
-    const/4 v0, 0x0
-
-    invoke-virtual {p0, v0}, Landroid/os/UserManager;->getUsers(Z)Ljava/util/List;
-
-    move-result-object p0
-
-    invoke-interface {p0}, Ljava/util/List;->iterator()Ljava/util/Iterator;
-
-    move-result-object p0
-
-    :cond_0
-    invoke-interface {p0}, Ljava/util/Iterator;->hasNext()Z
-
-    move-result v1
-
-    if-eqz v1, :cond_1
-
-    invoke-interface {p0}, Ljava/util/Iterator;->next()Ljava/lang/Object;
-
-    move-result-object v1
-
-    check-cast v1, Landroid/content/pm/UserInfo;
-
-    if-eqz v1, :cond_0
-
-    iget v1, v1, Landroid/content/pm/UserInfo;->id:I
-
-    invoke-static {v1}, Lcom/samsung/android/knox/SemPersonaManager;->isDarDualEncryptionEnabled(I)Z
-
-    move-result v1
-
-    if-eqz v1, :cond_0
-
-    const/4 v0, 0x1
-
-    :cond_1
-    const-string p0, "KnoxUtils"
-
-    const-string v1, "isDualDAREnabled result : "
-
-    invoke-static {v1, p0, v0}, Lcom/android/settings/AirplaneModeEnabler$$ExternalSyntheticOutline0;->m(Ljava/lang/String;Ljava/lang/String;Z)V
-
-    return v0
-
-    :cond_2
-    new-instance p0, Ljava/lang/IllegalStateException;
-
-    const-string v0, "Unable to load UserManager"
-
-    invoke-direct {p0, v0}, Ljava/lang/IllegalStateException;-><init>(Ljava/lang/String;)V
-
-    throw p0
+    const/4 p0, 0x0
+
+    return p0
diff --git a/smali_classes5/com/samsung/android/settings/lockscreen/LockScreenSettings.smali b/smali_classes5/com/samsung/android/settings/lockscreen/LockScreenSettings.smali
--- a/smali_classes5/com/samsung/android/settings/lockscreen/LockScreenSettings.smali
+++ b/smali_classes5/com/samsung/android/settings/lockscreen/LockScreenSettings.smali
@@ -72,3 +72 @@
-    invoke-static {}, Lcom/samsung/android/knox/dar/ddar/DualDarManager;->isOnDeviceOwnerEnabled()Z
-
-    move-result v0
+    const/4 v0, 0x0
diff --git a/smali_classes5/com/samsung/android/settings/lockscreen/LockUtils.smali b/smali_classes5/com/samsung/android/settings/lockscreen/LockUtils.smali
--- a/smali_classes5/com/samsung/android/settings/lockscreen/LockUtils.smali
+++ b/smali_classes5/com/samsung/android/settings/lockscreen/LockUtils.smali
@@ -1910,3 +1910 @@
-    invoke-static {}, Lcom/samsung/android/knox/dar/ddar/DualDarManager;->isOnDeviceOwnerEnabled()Z
-
-    move-result v1
+    const/4 v1, 0x0
@@ -2090,3 +2088 @@
-    invoke-static {}, Lcom/samsung/android/knox/dar/ddar/DualDarManager;->isOnDeviceOwnerEnabled()Z
-
-    move-result v5
+    const/4 v5, 0x0
diff --git a/smali_classes5/com/samsung/android/settings/lockscreen/controller/DualDarDoScreenLockTypePreferenceController.smali b/smali_classes5/com/samsung/android/settings/lockscreen/controller/DualDarDoScreenLockTypePreferenceController.smali
--- a/smali_classes5/com/samsung/android/settings/lockscreen/controller/DualDarDoScreenLockTypePreferenceController.smali
+++ b/smali_classes5/com/samsung/android/settings/lockscreen/controller/DualDarDoScreenLockTypePreferenceController.smali
@@ -353,3 +353 @@
-    invoke-static {}, Lcom/samsung/android/knox/dar/ddar/DualDarManager;->isOnDeviceOwnerEnabled()Z
-
-    move-result v0
+    const/4 v0, 0x0
diff --git a/smali_classes5/com/samsung/android/settings/password/LockTypePolicy.smali b/smali_classes5/com/samsung/android/settings/password/LockTypePolicy.smali
--- a/smali_classes5/com/samsung/android/settings/password/LockTypePolicy.smali
+++ b/smali_classes5/com/samsung/android/settings/password/LockTypePolicy.smali
@@ -297,3 +297 @@
-    invoke-static {}, Lcom/samsung/android/knox/dar/ddar/DualDarManager;->isOnDeviceOwnerEnabled()Z
-
-    move-result p2
+    const/4 p2, 0x0
--
2.51.1
